services/federator/test/integration/Test/Federator/InwardSpec.hs

```diff
@@ -109,14 +111,19 @@ inwardBrigCall requestPath payload = do
           { GRPC.component = Brig,
             GRPC.path = requestPath,
             GRPC.body = LBS.toStrict payload,
-            GRPC.originDomain = "foo.example.com"
+            GRPC.originDomain = "localhost.example.com"
           }
   liftIO $ gRpcCall @'MsgProtoBuf @Inward @"Inward" @"call" c brigCall
 
 viewFederatorExternalClient :: (MonadIO m, MonadHttp m, MonadReader TestEnv m, HasCallStack) => m GrpcClient
 viewFederatorExternalClient = do
   Endpoint fedHost fedPort <- cfgFederatorExternal . view teTstOpts <$> ask
-  client <- createGrpcClient (grpcClientConfigSimple (Text.unpack fedHost) (fromIntegral fedPort) False)
+  exampleCert <- liftIO $ BS.readFile "test/resources/unit/localhost.pem"
+  let cfg =
+        (grpcClientConfigSimple (Text.unpack fedHost) (fromIntegral fedPort) False)
+          { _grpcClientConfigHeaders = [("X-SSL-Certificate", exampleCert)]
```

Comment:
It crossed my mind, but I am not completely sure about this: should the headers list be expanded here rather than replaced?

Reply:
The default is empty, so it makes no difference here.